spec/jobs/journaled/delivery_job_spec.rb

```diff
@@ -71,63 +79,47 @@
     end
 
     context 'when the stream name is not set' do
-      let(:stream_name) { nil }
+      let(:args) { [{ serialized_event: '{"foo":"bar"}', partition_key: 'fake_partition_key', stream_name: nil }] }
 
-      it 'raises an KeyError error' do
-        expect { described_class.perform_now(**args) }.to raise_error ArgumentError, 'missing keyword: stream_name'
+      it 'raises an ArgumentError error' do
+        expect { described_class.perform_now(*args) }.to raise_error ArgumentError, /missing keyword: :?stream_name/
       end
     end
 
-    unless Gem::Version.new(Journaled::VERSION) < Gem::Version.new('5.0.0')
+    unless Gem::Version.new(Journaled::VERSION) < Gem::Version.new('6.0.0')
       raise <<~MSG
-        Hey! I see that you're bumping the version to 5.0!
+        Hey! I see that you're bumping the version to 6.0!
 
         This is a reminder to:
-        - remove the `app_name` argument (and related logic) from `Journaled::DeliveryJob`,
-        - remove the following app_name test contexts, and
-        - make `stream_name` a required kwarg
+        - Remove the `legacy_kwargs` argument (and related logic) from `Journaled::DeliveryJob`
+        - Remove the related test context below
 
         Thanks!
       MSG
     end
 
-    context 'when the legacy app_name argument is present but nil' do
-      let(:args) { { serialized_event: serialized_event, partition_key: partition_key, app_name: nil } }
-
-      around do |example|
-        with_env(JOURNALED_STREAM_NAME: 'legacy_stream_name') { example.run }
-      end
+    context 'when supplying legacy kwargs (a single event) instead of a list of events' do
```

Comment:
Covering the legacy kwarg case for the time being (see the version check above, to remind us to remove this support when we hit version 6.0)